### PR TITLE
✅ Test TestParametersPlugin.java

### PR DESCRIPTION
**Pull request by aldebaran30701**
Created a test class and some unit tests for the methods.

As this is an experimental plugin I have not tested to 100%. I've attempted to get the main functionality tested and a few tests of key parameters.

I did not execute the plugin with the pluginexecution rather tested each method similar to how `MergeTransactionsNGTest.java` does.

Applicable tickets:
#413

### CODE REVIEW

**Review by arcturus2 — APPROVED** [review]
Nice work @aldebaran30701 there are 16 tests here and this is really helpful to make sure that we maintain the correct behavior that is used to demonstrate what is possible with plugin parameters.

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+        // Intentionally left blank
+    }
+
+    /**
+     * Test of getType method, of class TestParametersPlugin.
+     */
+    @Test
+    public void testGetType() {
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java:80]: Just a comment but I reckon the likes of getType, getPosition, and getDescription probably don't need a test as they're one line functions that return the same value everytime (so no real value added from having these unit tests)

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+        // Set plugin query name here before execution
+        result.getParameters().get(CoreGlobalParameters.QUERY_NAME_PARAMETER_ID).setStringValue("TESTPARAMETERSPLUGIN");
+
+        // Set plugin parameters here before execution
+        result.getParameters().get(TestParametersPlugin.LEVEL_PARAMETER_ID).setStringValue("None");
+
+        try {
+            instance.query(recordStore, interaction, result);
+        } catch (final PluginException ex) {
+            fail();
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java:313]: I could be wrong but if you're throwing the exception then that should be enough for the unit test to fail without needing this function

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+            assertEquals(queryResults.get(GraphRecordStoreUtilities.SOURCE + AnalyticConcept.VertexAttribute.COMMENT), "TESTPARAMETERSPLUGIN");
+            assertEquals(queryResults.get(GraphRecordStoreUtilities.SOURCE + TemporalConcept.VertexAttribute.LAST_SEEN), DateTimeFormatter.ISO_INSTANT.format(dtrStartEnd[0]).replace("Z", ".000Z"));
+
+            assertEquals(queryResults.get(GraphRecordStoreUtilities.DESTINATION + AnalyticConcept.VertexAttribute.RAW), "name2@domain2.com");
+            assertEquals(queryResults.get(GraphRecordStoreUtilities.DESTINATION + AnalyticConcept.VertexAttribute.TYPE), "Email");
+            assertEquals(queryResults.get(GraphRecordStoreUtilities.DESTINATION + AnalyticConcept.VertexAttribute.COMMENT), "TESTPARAMETERSPLUGIN");
+            assertEquals(queryResults.get(GraphRecordStoreUtilities.DESTINATION + TemporalConcept.VertexAttribute.LAST_SEEN), DateTimeFormatter.ISO_INSTANT.format(dtrStartEnd[1]).replace("Z", ".000Z"));
+
+        } catch (final PluginException ex) {
+            fail();
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java:358]: similar here

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+        // Set plugin query name here before execution
+        result.getParameters().get(CoreGlobalParameters.QUERY_NAME_PARAMETER_ID).setStringValue("TESTPARAMETERSPLUGIN");
+
+        // Set plugin parameters here before execution
+        result.getParameters().get(TestParametersPlugin.CRASH_PARAMETER_ID).setBooleanValue(false);
+
+        try {
+            instance.query(recordStore, interaction, result);
+        } catch (final PluginException ex) {
+            fail();
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java:386]: and here

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java (hunk 1 above)
Comment (aldebaran30701) [reply CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java:80]: Would you like me to remove them? They were autogenerated and I decided to spend the few seconds implementing something while it was there.

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+
+        // Set plugin query name here before execution
+        result.getParameters().get(CoreGlobalParameters.QUERY_NAME_PARAMETER_ID).setStringValue("TESTPARAMETERSPLUGIN");
+
+        // Set plugin parameters here before execution
+        result.getParameters().get(TestParametersPlugin.CRASH_PARAMETER_ID).setBooleanValue(true);
+
+        try {
+            instance.query(recordStore, interaction, result);
+        } catch (final PluginException ex) {
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java:413]: if the unit test throws Exception, you won't need a try-catch here (if all you're doing in the catch is throwing the exception anyway)

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Review by antares1470 — APPROVED** [review]

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java (hunk 1 above)
Comment (antares1470) [reply CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java:79]: They're fine to be left there. They're not doing any damage really

**Pull request merged**